File: my-folder/0206-reverse-linked-list/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp = head
        prev = None
        while temp:
            curr = temp
            temp = temp.next
            curr.next = prev
            prev = curr
        
        return prev
        
        

File: my-folder/0206-reverse-linked-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode curr = head;
        ListNode prev = null;
        ListNode dummy = head;
        while(dummy != null){
            curr = dummy;
            dummy = dummy.next;
            curr.next = prev;
            prev = curr;
        }
        return prev;        
    }
}

// 1 -> 2 -> 3 -> 4 -> 5
// prev = null
// curr = head



